metadata-ingestion/src/datahub/cli/generate_report/report_generator.py

```diff
@@ -0,0 +1,247 @@
+        config = GenerateReportConfig.parse_obj(config_dict)
+        return cls(config)
+
+class PrivacyTermExtractor:
+    DATAHUB_GRAPHQL_ENDPOINT: str = '/api/graphql'
+    PAGE_SIZE: int = 500
+    MAX_SEARCH_RESULTS_TOTAL: int = 100000
+    GRAPHQL_QUERY: str = """
+    query GetDatasetGlossaryTerms($search_query: String!, $page_size: Int, $offset:Int) {
+      search(input: {
+        type: DATASET,
+        # search query cannot be empty, use "*" or by searching for a specific dataPlatoform name e.g. snowflake
+        query: $search_query,
+        count: $page_size,
+        start: $offset,
+      }) {
+        total
+        searchResults {
+          entity {
+            ... on Dataset {
+              urn
+              editableSchemaMetadata {
+                editableSchemaFieldInfo {
+                  fieldPath
+                  glossaryTerms {
+                    terms {
+                      term {
+                        urn
+                        name
+                      }
+                    }
+                  }
+                }
+              }
+              # zero represents the latest version
+              schemaMetadata(version: 0) {
+                name
+                fields {
+                  fieldPath
+                  glossaryTerms {
+                    terms {
+                      term {
+                        urn
+                        name
+                      }
+                    }
+                  }
+                }
+              }
+            }
+          }
+        }
+      }
+    }
+    """
+    graphql_api_url: str
+
+    def __init__(self, datahub_base_url: str) -> None:
+        self.graphql_api_url = f'{datahub_base_url}{self.DATAHUB_GRAPHQL_ENDPOINT}'
+
+    def yield_search_results(self, search_queries: Iterable[str]) -> Generator[Dict, None, None]:
+        return itertools.chain.from_iterable(
+            self._yield_rows(entity)
+            for search_query in search_queries
+            for entity in self._yield_graphql_search_results(search_query)
+        )
+
+    def _yield_graphql_search_results(self, search_query: str) -> Generator[Dict, None, None]:
+        total = self.MAX_SEARCH_RESULTS_TOTAL
+        offset = 0
+        while offset < total:
+            graphql_variables = {
+                'search_query': search_query,
+                'page_size': self.PAGE_SIZE,
+                'offset': offset
+            }
+            logger.info(f'Issuing graphql query with variables: {graphql_variables}')
+
+            payload = {'query': self.GRAPHQL_QUERY, 'variables': graphql_variables}
+            response = requests.post(self.graphql_api_url, json=payload)
+            response.raise_for_status()
+            response_json = response.json()
+            for entity in response_json['data']['search']['searchResults']:
+                yield entity
+
+            total = response_json['data']['search']['total']
```

Comment:
just to confirm, `total` here won't just be the # of items returned for the current page, but rather the sum of all previous items as well?

also nit: another `total` defined in this scope, might be clearer to name this something different?

Reply:
ya `total` will be the sum of the whole query (basically assuming theres no pages), `total` of the `search` query.

The `total` is actually supposed to be overwritten, it's default to `MAX_SEARCH_RESULTS_TOTAL` so that `offset < total` will be true for the first loop.

👍  I can rename the variable to be clearer though.

Reply:
renamed to `total_num_of_datasets` and also made the base condition `None` so it's more clear, i got rid of `MAX_SEARCH_RESULTS_TOTAL`.
https://github.com/Affirm/datahub/pull/3/commits/704544a4f8742b1afdffc30a5b94e8514ff0b7b4